### PR TITLE
bump kindnetd deps and remove pkg/errors dep

### PR DESCRIPTION
**Pull request by aojea**
- remove pkg/errors dependency
- bump golang to 1.16
- bump k8s deps to 1.21.2
- bump base image to debian-iptables:buster-v1.6.5
- bump klog ro 2.9.0

### CODE REVIEW

**Comment by aojea** [comment]
/hold
I have to build the image and publish it

File: images/kindnetd/go.mod [hunk 1]
```diff
@@ -4,11 +4,10 @@ go 1.13
...
 require (
 	github.com/coreos/go-iptables v0.4.5
-	github.com/pkg/errors v0.9.1
 	github.com/vishvananda/netlink v1.1.0
 	github.com/vishvananda/netns v0.0.0-20200520041808-52d707b772fe // indirect
-	k8s.io/api v0.20.5
-	k8s.io/apimachinery v0.20.5
-	k8s.io/client-go v0.20.5
-	k8s.io/klog/v2 v2.4.0
+	k8s.io/api v0.21.2
```
Comment (BenTheElder) [inline images/kindnetd/go.mod:9]: time to bump go in the image?

Comment (BenTheElder) [reply images/kindnetd/go.mod:9]: and probably the `go` statement in the module?

File: images/kindnetd/Dockerfile [hunk 2]
```diff
@@ -24,6 +24,6 @@ COPY . .
 RUN CGO_ENABLED=0 go build -o ./kindnetd ./cmd/kindnetd
 
 # build real kindnetd image
-FROM k8s.gcr.io/build-image/debian-iptables:buster-v1.6.3
+FROM k8s.gcr.io/build-image/debian-iptables:buster-v1.6.5
```
Comment (BenTheElder) [inline images/kindnetd/Dockerfile:27]: confirmed this is latest.

**Review by BenTheElder — APPROVED** [review]
/hold
/lgtm
/approve

**Pull request merged**